### PR TITLE
Adds ETC2 and ASTC types to enable testing when relevant.

### PR DESCRIPTION
**Pull request by lokokung**
- Modifies texture sizes to be functions of the block sizes to generalize the tests.

Tests pass on dev build of Chromium with changes from https://chromium-review.googlesource.com/c/chromium/src/+/3219513 patched in on an Intel-Linux-Vulkan environment. (Currently this is one of the only environment combinations with the support enabled/available.)

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [x] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [x] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [x] Existing (or new) test helpers are used where they would reduce complexity.
- [x] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 1]
```diff
@@ -665,40 +665,91 @@ g.test('color_textures,compressed,non_array')
...
-          srcTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 64, height: 52, depthOrArrayLayers: 1 },
-        },
-        // None of the widths or heights are power of 2
-        {
-          srcTextureSize: { width: 60, height: 52, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 60, height: 52, depthOrArrayLayers: 1 },
-        },
-      ])
+      .expand('textureSize', p => {
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:668]: Perhaps change the parameter to be `textureSizeInBlocks` and multiply-in the block size inside the test function? I think this is preferable in places where we don't care to test non-block-sized values (like this one). Plus it means that on the off-chance we add formats with 3d blocks (like 4x4x2) later, it'll be more straightforward to update

**Review by kainino0x — APPROVED** [review]
lgtm!

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts (hunk 1 above)
Comment (lokokung) [reply src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts]: Done!

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 2]
```diff
@@ -665,51 +665,42 @@ g.test('color_textures,compressed,non_array')
     u
       .combine('format', kCompressedTextureFormats)
       .beginSubcases()
-      .combine('textureSize', [
+      .combine('textureSizeInBlocks', [
         // The heights and widths are all power of 2
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:669]:
```suggestion
        // The heights and widths in blocks are all power of 2
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 3]
```diff
@@ -665,51 +665,42 @@ g.test('color_textures,compressed,non_array')
...
-        {
-          srcTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-        },
+        { src: { width: 16, height: 8 }, dst: { width: 16, height: 8 } },
         // The virtual width of the source texture at mipmap level 2 (15) is not a multiple of 4
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:671]:
```suggestion
        // The virtual width of the source texture at mipmap level 2 (15) is not a multiple of 4 blocks
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 4]
```diff
@@ -665,51 +665,42 @@ g.test('color_textures,compressed,non_array')
...
-        {
-          srcTextureSize: { width: 60, height: 32, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-        },
+        { src: { width: 15, height: 8 }, dst: { width: 16, height: 8 } },
         // The virtual width of the destination texture at mipmap level 2 (15) is not a multiple
         // of 4
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:674]:
```suggestion
        // of 4 blocks
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 5]
```diff
@@ -665,51 +665,42 @@ g.test('color_textures,compressed,non_array')
...
-        {
-          srcTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 60, height: 32, depthOrArrayLayers: 1 },
-        },
+        { src: { width: 16, height: 8 }, dst: { width: 15, height: 8 } },
         // The virtual height of the source texture at mipmap level 2 (13) is not a multiple of 4
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:676]:
```suggestion
        // The virtual height of the source texture at mipmap level 2 (13) is not a multiple of 4 blocks
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 6]
```diff
@@ -665,51 +665,42 @@ g.test('color_textures,compressed,non_array')
...
-        {
-          srcTextureSize: { width: 64, height: 52, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-        },
+        { src: { width: 16, height: 13 }, dst: { width: 16, height: 8 } },
         // The virtual height of the destination texture at mipmap level 2 (13) is not a
         // multiple of 4
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:679]:
```suggestion
        // multiple of 4 blocks
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 7]
```diff
@@ -665,51 +665,42 @@ g.test('color_textures,compressed,non_array')
...
-        {
-          srcTextureSize: { width: 64, height: 32, depthOrArrayLayers: 1 },
-          dstTextureSize: { width: 64, height: 52, depthOrArrayLayers: 1 },
-        },
+        { src: { width: 16, height: 8 }, dst: { width: 16, height: 13 } },
         // None of the widths or heights are power of 2
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:681]:
```suggestion
        // None of the widths or heights in blocks are power of 2
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 8]
```diff
@@ -769,30 +760,32 @@ g.test('color_textures,compressed,array')
     u
       .combine('format', kCompressedTextureFormats)
       .beginSubcases()
-      .combine('textureSize', [
+      .combine('textureSizeInBlocks', [
         // The heights and widths are all power of 2
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:764]:
```suggestion
        // The heights and widths in blocks are all power of 2
```

File: src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts [hunk 9]
```diff
@@ -769,30 +760,32 @@ g.test('color_textures,compressed,array')
...
-        {
-          srcTextureSize: { width: 8, height: 8, depthOrArrayLayers: 5 },
-          dstTextureSize: { width: 8, height: 8, depthOrArrayLayers: 5 },
-        },
+        { src: { width: 2, height: 2 }, dst: { width: 2, height: 2 } },
         // None of the widths or heights are power of 2
```
Comment (kainino0x) [inline src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts:766]:
```suggestion
        // None of the widths or heights in blocks are power of 2
```

**Review by kainino0x — APPROVED** [review]
lgtm, fixed up some comments that were already wrong before this change

**Comment by lokokung** [comment]
Probably blocked until https://chromium-review.googlesource.com/c/chromium/src/+/3219513 is rolled into Chrome?

**Comment by kainino0x** [comment]
@lokokung that's a Chromium change, so it's already in - did you mean to link a Dawn change? Or maybe you meant a chrome release - we don't have to wait for those. In general we don't have to wait for anything to land changes here since the chromium webgpu gardener can just suppress failures, but it can be convenient to delay slightly so they don't have to do that.

**Comment by lokokung** [comment]
Ah, oops, I misunderstood the issue here. Updated the version for @webgpu/types which should fix the issue with the last build.

**Pull request merged**